src/Steeltoe.Extensions.Logging.SerilogDynamicLogger/SerilogDynamicProvider.cs

```diff
@@ -0,0 +1,224 @@
+
+                        results[name] = config;
+                    }
+                }
+            }
+
+            return results.Values;
+        }
+
+        public void SetLogLevel(string category, LogLevel? level)
+        {
+            var filteredPairs = _loggerSwitches.Where(kvp => kvp.Key.StartsWith(category));
+            foreach (var kvp in filteredPairs)
+            {
+                var currentLevel = level ?? GetConfiguredLevel(kvp.Key);
+                if (currentLevel != null)
+                {
+                    kvp.Value.MinimumLevel = ToSerilogLevel(currentLevel);
+                }
+            }
+        }
+
+        private LogLevel? GetConfiguredLevel(string name)
+        {
+            LogLevel? returnValue = null;
+            if (name == "Default")
+            {
+                var defaultValue = _serilogOptions.MinimumLevel.Default;
+                returnValue = FromSerilogLevel(defaultValue);
+            }
+            else
+            {
+                var overrides = _serilogOptions.MinimumLevel.Override;
+                if (overrides != null
+                    && overrides.ContainsKey(name)
+                    && overrides.TryGetValue(name, out LogEventLevel configuredLevel))
+                {
+                    returnValue = FromSerilogLevel(configuredLevel);
+                }
+            }
+
+            return returnValue;
+        }
+
+        private LogLevel GetEffectiveLevel(string name)
+        {
+            LoggingLevelSwitch levelSwitch;
+            _loggerSwitches.TryGetValue(name, out levelSwitch);
+            return FromSerilogLevel(levelSwitch.MinimumLevel);
+        }
+
+        private IEnumerable<string> GetKeyPrefixes(string name)
+        {
+            while (!string.IsNullOrEmpty(name))
+            {
+                yield return name;
+                var lastIndexOfDot = name.LastIndexOf('.');
+                if (lastIndexOfDot == -1)
+                {
+                    yield return "Default";
+                    break;
+                }
+
+                name = name.Substring(0, lastIndexOfDot);
+            }
+        }
+
+        private LogLevel FromSerilogLevel(LogEventLevel level)
+        {
+            switch (level)
+            {
+                case LogEventLevel.Verbose: return LogLevel.Trace;
+                case LogEventLevel.Debug: return LogLevel.Debug;
+                case LogEventLevel.Information: return LogLevel.Information;
+                case LogEventLevel.Warning: return LogLevel.Warning;
+                case LogEventLevel.Error: return LogLevel.Error;
+                case LogEventLevel.Fatal: return LogLevel.Critical;
+                default: return LogLevel.None;
+            }
+        }
+
+        private LogEventLevel ToSerilogLevel(LogLevel? level)
+        {
+            if (level == null)
+            {
+                return LogEventLevel.Fatal;
+            }
+
+            switch (level.Value)
```

Comment:
Would this accomplish the same result? `return (LogEventLevel)level.Value;`

As I'm reading the switch, I think the one difference might be with LogLevel.None, but you could add that to the `if` above

Reply:
Yes that would be the same. And we can use the if for none.